src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+                    else destFile.createNewFile();
+                    FileInputStream fin = new FileInputStream(source);
+                    FileOutputStream fout = new FileOutputStream(dest);
+                    int i;
+                    do {
+                        i = fin.read();
+                        if (i != -1)
+                            fout.write(i);
+                    }
+                    while (i != -1);
+                    fin.close();
+                    fout.close();
+                } catch (IOException n) {
+                }
+                break;
+            case BUFFERED:
+                copy(source, dest);
+                break;
+        }
+    }
+
+    @Override
+    public void copy(String source, String dest) throws
+            FileNotFoundException, IllegalArgumentException, FileAlreadyExistException, FileCopyException {
+        try {
+            File sourceFile = new File(source);
+            File destFile = new File(dest);
+            if (!(sourceFile.exists())) throw new FileNotFoundException();
+            if (sourceFile.isDirectory()) throw new IllegalArgumentException();
+            if (destFile.exists()) throw new FileAlreadyExistException(destFile);
+            else destFile.createNewFile();
+            BufferedInputStream fin = new BufferedInputStream(new FileInputStream(source));
+            BufferedOutputStream fout = new BufferedOutputStream(new FileOutputStream(dest));
+            int i;
+            do {
+                i = fin.read();
+                if (i != -1)
+                    fout.write(i);
+            }
+            while (i != -1);
+            fin.close();
+            fout.close();
+            System.out.println("File copied successfully!!");
+        } catch (IOException n) {
+        }
+    }
+
+
+    @Override
+    public File currentDirectory() {
+        return currentDirectory;
+    }
+
+    public void setCurrentDirectory(String newPath) {
+        this.currentDirectory = new File(newPath);
+
+    }
+
+    private String FilesInfo(List<File> files) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < files.size(); i++) {
+            sb.append(fileInfo(files.get(i)));
+        }
+        return sb.toString();
+
+    }
+
+    private FileFilter fileFilterForAll() {
+        FileFilter fileFilterForAll = new FileFilter() {
+            @Override
+            public boolean accept(File pathname) {
+                return true;
+            }
+        };
+        return fileFilterForAll;
+    }
+
+
+    Comparator comparatorForLexAscending = new Comparator<File>() {
+        @Override
+        public int compare(File o1, File o2) {
+            if (o1.isDirectory())
+                return o2.isDirectory() ? o1.compareTo(o2) : -1;
+            else if (o2.isDirectory())
+                return 1;
+            return o1.compareTo(o2);
+        }
+
+    };
+    Comparator comparatorForLexAscendingNoOrder = new Comparator<File>() {
```

Comment:
Do use types for your declarations and avoid raw types.
Your declaration should be:
```
private final Comparator<File> field_name = new Comparator<File>(){
// implementation
}
```